cpp/src/gandiva/gdv_string_function_stubs.cc

```diff
@@ -264,6 +265,304 @@ const char* gdv_fn_lower_utf8(int64_t context, const char* data, int32_t data_le
   return out;
 }
 
+// converts utf8 to utf16 encoding
+GANDIVA_EXPORT
+const char* gdv_fn_encode(int64_t context, const char* data, int32_t data_len,
+                          const char* charset, int32_t charset_len, int32_t* out_len) {
+  if (data_len == 0) {
+    *out_len = 0;
+    return "";
+  }
+
+  char* out =
+      reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, 32 * data_len));
+  if (out == nullptr) {
+    gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
+    *out_len = 0;
+    return "";
+  }
+
+  int32_t char_len;
+  uint32_t char_codepoint;
+  int32_t y = 5;
+  int32_t z = 9;
+
+  std::string charset_str(charset, charset_len);
+
+  if (boost::iequals(charset_str, "UTF-16BE")) {
```

Comment:
I may be misunderstanding, but is this function really checking the charset value for all individual inputs? Isn't it completely wasteful? What is the point of a JIT compiler if you end up doing this?

Reply:
This isn't answering the question.

Reply:
Why do you think it's wasteful?

Reply:
Because for every individual input value, we're paying the cost of case-insensitive comparison. The overhead will be particular high for short input strings, which I presume are very common.

Reply:
Well, those could be separate functions (one for utf-16-le, one for utf-16-be) instead of taking a "charset" parameter? That would sound better IMHO (though the Gandiva maintainers would have to make a decision about this, not me :-)).

Reply:
Is there any example like this function which has two or more sub-functionalities still two or more different functions are made for it. Can you tell?

Reply:
Why not I am interested ofcourse :-) but at the same time I am trying to understand the issue also.